cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -347,10 +355,29 @@ private void hbaseDelete(List<Delete> deletes) throws IOException {
   protected Scanner scanPersisted(co.cask.cdap.api.dataset.table.Scan scan) throws Exception {
     ScanBuilder hScan = tableUtil.buildScan();
     hScan.addFamily(columnFamily);
-    // todo: should be configurable
-    // NOTE: by default we assume scanner is used in mapreduce job, hence no cache blocks
-    hScan.setCacheBlocks(false);
-    hScan.setCaching(1000);
+
+    // TODO (CDAP-11954): use common utility method to extract these configs
+    if (scan.getProperties().containsKey(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)) {
+      hScan.setCacheBlocks(Boolean.valueOf(scan.getProperties().get(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)));
+    } else if (arguments.containsKey(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)) {
+      hScan.setCacheBlocks(Boolean.valueOf(arguments.get(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)));
+    } else if (properties.containsKey(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)) {
+      hScan.setCacheBlocks(Boolean.valueOf(properties.get(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)));
+    } else {
+      // NOTE: by default we assume scanner is used in mapreduce job, hence no cache blocks
+      hScan.setCacheBlocks(false);
+    }
+
+    if (scan.getProperties().containsKey(CONFIG_HBASE_CLIENT_SCANNER_CACHING)) {
+      hScan.setCaching(Integer.valueOf(scan.getProperties().get(CONFIG_HBASE_CLIENT_SCANNER_CACHING)));
+    } else if (arguments.containsKey(CONFIG_HBASE_CLIENT_SCANNER_CACHING)) {
+      hScan.setCaching(Integer.valueOf(arguments.get(CONFIG_HBASE_CLIENT_SCANNER_CACHING)));
+    } else if (properties.containsKey(CONFIG_HBASE_CLIENT_SCANNER_CACHING)) {
+      hScan.setCaching(Integer.valueOf(properties.get(CONFIG_HBASE_CLIENT_SCANNER_CACHING)));
+    } else {
+      // NOTE: by default we use this hard-coded value, for backwards-compatibility with CDAP<4.1.2|4.2.1|4.3
```

Comment:
Yes, and in that case we were hard-coding the value of 1000, instead of defaulting to what's in the hConf. I am not sure whether that was a good idea, but changing it now may have performance impact on existing apps, so I am keeping it for backward-compatibility.